[docs] improve router docs (#25002)
expo/expo: docs/pages/router/reference/hooks.mdx

}
```

> For nested layouts, you can pass a secondary argument, denoting the layout route, to access higher-order layouts. For example:

Reviewer: ```suggestion
When using nested layouts, you can access higher-order layouts by passing a secondary argument denoting the layout route. For example:
```

it was a bit difficult to read with too much pause (`,` usage). So I tried to make it sound a bit linear.

<FileTree
  files={[
    ['app/_layout.js', <code>useNavigation('/')</code>],
    ['app/orders/_layout.js', <code>useNavigation('/orders')</code>],
    ['app/orders/menu/_layout.js', <code>useNavigation('/orders/menu')</code>],
  ]}
/>

> ```tsx app/orders/menu/index.tsx

Reviewer: ````suggestion
{/* prettier-ignore */}
```tsx app/orders/menu/index.tsx
````

import { useNavigation } from 'expo-router';

export default function MenuRoute() {
  const rootLayout = useNavigation('/');

  const ordersLayout = useNavigation('/orders');

  // Same as the default results of `useNavigation()` when invoked in this route.
  const parentLayout = useNavigation('/orders/menu');

>   // ...

Reviewer: ```suggestion
  /* @hide ... */ /* @end */
```

}
```

> If you attempt to access a layout that doesn't exist, an error `Could not find parent navigation with route "/non-existent".` will be thrown.

Reviewer: ```suggestion
If you attempt to access a layout that doesn't exist, an error such as `Could not find parent navigation with route "/non-existent".` is thrown.
```
